specification/under_development/profiles/Adaptive_Profile.md

```diff
@@ -0,0 +1,1028 @@
+# Adaptive Profile
+
+This profile defines a subset of the QIR specification to support a coherent set
+of functionalities and capabilities that might be offered by a quantum backend.
+Like all profile specifications, this document is primarily intended for
+[compiler backend](https://en.wikipedia.org/wiki/Compiler#Back_end) authors as
+well as contributors to the [targeting
+stage](../Compilation_And_Targeting.md#targeting) of the QIR compiler.
+
+The adaptive profile specifies supersets of the base profile that enable
+control flow based on mid-circuit measurements and classical computations
+within coherence times. A backend can support this profile by supporting
+a minimum set of features beyond the base profile and can opt in for features
+beyond that. This marks the first attempt at defining a profile greater in scope
+than the base profile but still limited when compared to the full QIR specification.
+
+Being a superset of the base profile means that these 3 *capabilities* must be
+implemented (a few restrictions are removed from the base profile):
+
+1. An adaptive profile program can execute a sequence of quantum instructions
+   that transform the quantum state.
+2. An adaptive profile program can measure any qubit used in the program and
```

Comment:
I find this confusing as phrased. The phrasing makes the difference to the base profile not very clear. 
Suggested phrasing: "A backend must support applying measurement instructions at any point during execution without impacting the state of the qubits that are not measured. Specifically, `quantum__qis__mz__body` or some other measurement function must be supported in the quantum instruction set."